src/pages/App.tsx

```diff
@@ -16,6 +16,8 @@ import { useFaucet } from '@gooddollar/web3sdk-v2'
 import TransactionUpdater from '../state/transactions/updater'
```

Comment:
I see a lot of stuff been added to App, let's move it to separate component `src/components/walletchat/WalletChat.js`:

```javascript
import React from 'react'

import { WalletChatWidget } from 'react-wallet-chat'
import { useConnectWallet, useSetChain } from '@web3-onboard/react'
import { isMobile } from '../../lib/utils/platform'

const WalletChat = () => {
  const [{ chains, connectedChain }] = useSetChain()
   const [{ wallet }] = useConnectWallet()
   const connectedAccount = wallet?.accounts[0]
   const activeChain = chains.find((chain) => chain.id === connectedChain?.id)
   const chainId = activeChain?.token === 'CELO' ? 42_220 : 1

  return (
    <WalletChatWidget
         style={{ marginBottom: isMobile ? '75px' : '0px' }}
         connectedWallet={
             connectedAccount &&
             chainId && {
                 walletName: wallet.label,
                 account: connectedAccount.address,
                 chainId,
             }
         }
     />
  )
}

export default WalletChat

Reply:
also not sure you need for onboard and connect wallet. we have own wallet could be accessible through useWallet() hook

import { useWallet } from 'src/lib/wallet/GoodWalletProvider'

Reply:
useActiveWeb3React will return you 

```typescript
chainId: ChainId
label?: string
```

also it returns account - see AppBar.tsx
so it's possible to use this hook instead

Reply:
@johnsmith-gooddollar - This should be resolved, good suggestions!